### PR TITLE
refactor: move iso8601time and csttime to core/models/common

### PR DESCRIPTION
**Pull request by klesh**
### Summary
Currently, we have 2 `iso8601time` structs defined in `helpers/pluginhelper/api` and `helpers/utils` which is certainly a mistake. Another problem is both `iso8601time` and `csttime` were put into `helpers/api` which is not the right place for such commonly used structs. This PR fixes these problems.

1. delete `iso8601time` in the `helpers/utils`
2. move `iso8601time` and `csttime` from `helpers/pluginhelper/api` to `core/models/common` package
3. all **archived** models use plain `time.Time` instead of the `iso8601time` or `csttime` because they must be immutable and they don't need the features offered by these structs.

### CODE REVIEW

File: backend/core/models/common/iso8601time.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func init() {
 
 // Iso8601Time is type time.Time
 type Iso8601Time struct {
-	time   time.Time
+	Time   time.Time
```
Comment (d4x1) [inline backend/core/models/common/iso8601time.go:80]: If `Time` field is not private, then method `Iso8601Time.ToTime` will be duplicated. 
Can we delete it ?

File: backend/impls/dalgorm/db_mapper.go [hunk 2]
```diff
@@ -33,7 +34,7 @@ func ToDatabaseMap(tableName string, m map[string]any) map[string]any {
 			continue
 		}
 		if str, ok := v.(string); ok {
-			t, err := api.ConvertStringToTime(str)
+			t, err := common.ConvertStringToTime(str)
```
Comment (d4x1) [inline backend/impls/dalgorm/db_mapper.go:37]: `common` is under `models` package, `ConvertStringToTime` is a function, can we put it to `pluginhelper`?

**Review by d4x1 — COMMENTED** [review]
BTW, why don't we use time.RFC3339 directly ?

**Review by abeizn — APPROVED** [review]
lgtm

**Pull request merged**